lib/puppet/configurer.rb

```diff
@@ -56,7 +55,7 @@ def init_storage
     end
   end
 
-  def initialize(factory = Puppet::Configurer::DownloaderFactory.new, transaction_uuid = nil, job_id = nil)
+  def initialize(transaction_uuid = nil, job_id = nil)
```

Comment:
we aren't worried about this interface changing? It's not going to break things for users?

Reply:
I have minor concerns about it. Traditionally we'd do this in a Puppet 6. We could create a Puppet 6 branch now and retarget it there.

Reply:
@branan was similarly minorly concerned, but also said he considers this private API. His recommendation was to merge in 5.4.0 and revert if it causes issues somewhere, but I would also be fine saving this for Puppet 6 (though in that case I'd like to branch and still merge, so this doesn't get lost).